### PR TITLE
[Target] Add Target Parser for Arm(R) Cortex(R) M-Profile CPUs

### PR DESCRIPTION
**Pull request by Mousius**
This implements an initial Target Parser which uses the same logic as
the CMSIS-NN compiler flags to update the features and keys of the `c`
and `llvm` `Target`s.

Refactoring of the CMSIS-NN logic will be in a separate patch.

### CODE REVIEW

**Comment by Mousius** [comment]
cc @manupa-arm

File: tests/scripts/task_python_docs.sh [hunk 1]
```diff
@@ -81,6 +81,8 @@ IGNORED_WARNINGS=(
     'gen_gallery extension is not safe for parallel'
     'strategy:conv2d NHWC layout is not optimized for x86 with autotvm.'
     'strategy:depthwise_conv2d NHWC layout is not optimized for x86 with autotvm.'
+    'strategy:depthwise_conv2d with layout NHWC is not optimized for arm cpu.'
+    'strategy:dense is not optimized for arm cpu.'
```
Comment (lhutton1) [inline tests/scripts/task_python_docs.sh:85]: Curious, does this mean the `arm_cpu` key wasn't tested before this PR?

**Review by lhutton1 — APPROVED** [review]
LGTM! Just had one small question :)

File: tests/scripts/task_python_docs.sh (hunk 1 above)
Comment (Mousius) [reply tests/scripts/task_python_docs.sh:85]: yip, my guess is people didn't know to put `-device=arm_cpu` or `-keys=arm_cpu` so they were using x86 schedules 🙀

**Pull request merged**